### ISSUE TITLE
Empty xml report with checkstyle

### ISSUE THREAD

**Issue by Quantumplation** [palantir/tslint#1353]
### Bug Report

- __TSLint version__: `3.11.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: CLI

```
./node_modules/.bin/tslint src/**/*.ts --exclude **/*-spec.ts -o ./_lint/report.xml -t checkstyle
```

#### Actual behavior

File `report.xml` is created, but is empty

#### Expected behavior

File `report.xml` should have contents, something like:
```xml
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="4.3">
</checkstyle>
```

My build server fails because the produced file is invalid XML

**Comment by adidahiya** [comment]
Do you get any failures reported with other formatters? Try quoting your globs in the CLI (`'src/**/*.ts`, etc.). If the linter is getting files and reporting zero failures but not outputting any XML, this sounds like a bug ([see these tests](https://github.com/palantir/tslint/blob/master/test/formatters/checkstyleFormatterTests.ts)).

**Comment by Quantumplation** [comment]
If I introduce a linting violation, the command (unchanged) produces a valid checkstyle xml file, it's only when there are no linting errors in the project.

**Comment by caugner** [comment]
Same problem and expectation (cf. initial comment) here.

**Comment by JKillian** [comment]
This seems like a general TSLint w/ regards to formatters and not just an issue with the checkstyle one. [If there are no violations, nothing gets written to stdout.](https://github.com/palantir/tslint/blob/e23b1885fd8caca734c31078d70433640c422176/src/tslint-cli.ts#L254)

I'm not sure why this is the current behavior, I'm up for changing this unless anyone can think of a reason why it's better as it is now.

**Comment by caugner** [comment]
> I'm up for changing this unless anyone can think of a reason why it's better as it is now.

I agree with this change, because the current behavior makes the wrong assumption that formatters don't need to emit anything on success.

It should be noted that custom formatters could _potentially_ make use of this assumption and report a failure even on success (e.g. if the reporter prints _"There were N failures: (...)"_ and CI checks for the string `There were [0-9]+ failures`). As this is only a _hypothetical_ problem and the included standard formatters are not affected by this, it should suffice to note this in the release notes, so that users with this issue have a higher chance of quickly resolving it.

**Comment by caugner** [comment]
This issue [also affects the pmd reporter](https://github.com/palantir/tslint/blob/master/src/formatters/pmdFormatter.ts#L23).

**Comment by JKillian** [comment]
The exit status code provides a better way to indicate success/failure that the textual output of a formatter anyways. I'd say that the current behavior is broken, so let's fix this!

**Comment by Quantumplation** [comment]
Any update on this?

**Comment by JKillian** [comment]
This will be addressed by #1472 and will be fixed in the next release of TSLint
